components/portal/messages/filters.js

```diff
@@ -103,6 +103,36 @@ define(['angular'], function(angular) {
         });
       };
     })
+    .filter('filterByDates', function() {
+      return function(messages) {
+        // Get current date/time
+        var now = moment();
+        // Filter messages based on date/time
+        return messages.filter(function(message) {
+          if (message.goLiveDate || message.expireDate) {
```

Comment:
:confused: I'm a bit confused, `Array.prototype.filter` expects a boolean value.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter
This returns `message`, which happens to be `Object` which happens to be truthy.
AFAICT all paths return `message` so nothing will be filtered out.

Reply:
You're right about misusing `filter`. It would be more correct to just go with the good old `angular.forEach` in this case. 

However, not every path returns `message`. Cases that don't meet the criteria are not returned (thus not added to resulting array). For example: A notification that has both `goLiveDate` and `expireDate` specified, and `moment()` is not between those dates. If both dates are specified, it will stop at that block, not return `message`, and not proceed to the next conditional. If they're not both specified, it'll check first for `goLiveDate`. If that attribute is specified, we know there must not be an `expireDate` specified, so it's safe to only check whether the `goLiveDate` is in the past. If it's not specified, then we check for `expireDate`, at which point we know it's safe to only check against validity of that date.

The notification with id `sample-uportal-app-framework-expired-notification` is there to prove that this works, in lieu of a proper unit test.

Reply:
ahhh, sneaky
Moment comparisons return booleans so this can be flattened to

```js
// If both dates are defined, check between
if (message.goLiveDate && message.expireDate) {
  return now.isBetween(message.goLiveDate, message.expireDate)
}

// If goLive defined, check after
if (message.goLiveDate) {
  return now.isAfter(message.goLiveDate);
}

// If expire defined, check before
if (message.expireDate) {
  return now.isBefore(message.expireDate);
}

// Has neither date defined, so don't filter it
return true;
```